### DISCUSSION TITLE
error-stack: error building service with error-stack dependency

### DISCUSSION THREAD

**Discussion by AlexCzar** [hashintel/hash#1512] [error-stack (Help)] [answered]
Hi, I'm trying to use error-stack in a [Shuttle](https://docs.shuttle.rs) app. Could someone point me in the right direction in regards to the issue below?
```
error[E0514]: found crate `rustc_version` compiled by an incompatible version of rustc
 --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/error-stack-0.2.4/build.rs:1:5
  |
1 | use rustc_version::{version_meta, Channel, Version};
  |     ^^^^^^^^^^^^^
  |
  = note: the following crate versions were found:
          crate `rustc_version` compiled by rustc 1.63.0 (4b91a6ea7 2022-08-08): /opt/shuttle/shuttle-builds/rorschach/target/release/deps/librustc_version-9eecf20def6331b7.rlib
  = help: please recompile that crate using this compiler (rustc 1.67.0-nightly (11ebe6512 2022-11-01)) (consider running `cargo clean` first)
```
This might be a build environment misconfigurtion on the shuttle side, but maybe someone here is familiar with the error and able to quickly point out what to look into.
Shuttle uses rust 1.63.0, which should be fine for error-stack 0.2.4.

**Comment by indietyp** [comment, marked as answer]
Hi! You're encountering this error because you currently have a `target/` folder with packages compiled for `nightly-2022-11-01` instead of `1.63`. This might be because your default toolchain is `nightly`, while shuttle expects `1.63`. You should be able to fix this by simply running `cargo clean` before deploying via shuttle ☺️ 

Tip: You can pin the version of your binary/library by including a `rust-toolchain.toml` with the following contents:

```toml
[toolchain]
channel = "1.63"
components = ["clippy", "rustfmt"]
```

This will ensure that `cargo` commands will (regardless of the default system toolchain) run on `1.63`, and you won't get that error again in the future.

**Reply by AlexCzar** [reply]
This sounds right, but doesn't help, unfortunately.
My default toolchain is set to 1.63 via rustup, I've added the `rust-toolchain.toml` file with the suggested content, ran `cargo clean`, the error remains the same.
I think the nightly compiled stuff is not on my machine, but on Shuttle's.
Thanks for your input.
